### ISSUE TITLE
[docs] Display `aria-label` for components that should be internationalized

### ISSUE THREAD

**Issue by atomiks** [mui/base-ui#2245]
# Docs feedback

There are components like `Toast.Viewport` and `NumberField.Increment` that have default English `aria-label`s internally (in source), but the docs don't show that these should be internationalized. 

It's also possible we should remove the default English labels and defer it to the docs which makes it a lot more clear when looking at the examples that they need to be handled by the dev.

Details on auto ARIA translations: https://adrianroselli.com/2019/11/aria-label-does-not-translate.html#Update06

**Comment by mj12albert** [comment]
Makes sense ~ there could be more strings like these, e.g. `aria-roledescription` in number field
